### DISCUSSION TITLE
Creating tree sql from each object in first sql

### DISCUSSION THREAD

**Discussion by ahmed19992104** [adonisjs/core#1401] [Help]
how can i make querybuilder on each field returned from my first query
i tried it
const data = await Category.query().where('main_category_id' , params.search)

        data.forEach( async (element) => {

            data['main_categories'] = await Category.query().where('main_category_id' , element.category_id)

            data['main_categories'].forEach( async (element) => {

                data['sub_categories'] = await Category.query().where('main_category_id' , element.category_id)
                
            });
            
        });

**Comment by McSneaky** [comment]
`forEach` does not support async looping. Basically all loops that accept callback as a input don't work with async - await pattern

You can use for or while loops for it or wrap returning loop (like `.map()` into `Promise.all()`)

More about async looping in here: 
https://lavrton.com/javascript-loops-how-to-handle-async-await-6252dd3c795/

Sidenote: To get syntax highlight in GH you can use this:
\`\`\`LANG
CODE
\`\`\`
Where LANG is language you want highlight for and CODE is your code
\`\`\`js
console.log('Cool!');
\`\`\`
Translates to
```js
console.log('Cool!');
```
